src/standalone/UIKit/TextFieldWithHelp.tsx

```diff
@@ -1,51 +1,94 @@
-import React, { ForwardedRef } from "react";
+import React, { ForwardedRef, useCallback, useState } from "react";
 import {
 	TextField,
 	TextFieldProps,
 	IconButton,
 	InputAdornment,
-	InputAdornmentProps,
 } from "@material-ui/core";
-import { Info as InfoIcon } from "@material-ui/icons";
+import {
+	Info as InfoIcon,
+	HighlightOff as ClearIcon,
+} from "@material-ui/icons";
 import { InputLabelConfig, UIInputProps, useInputStyles } from "./CommonStyles";
 
 export interface TextFieldWithHelpProps extends UIInputProps {
 	/**
 	 * Optional callback which opens a dialog with information about the field
 	 */
 	openInfo?: () => void;
+	/**
+	 * Optional parameter which has formatted value to display on the textfield
+	 */
+	valueFormatted?: string;
+	/**
+	 * Optional callback which handles the clear value event
+	 */
+	onClear?: () => void;
 }
 
 const TextFieldWithHelp = React.forwardRef(function TextFieldWithHelpInner(
 	props: TextFieldWithHelpProps & TextFieldProps,
 	ref: ForwardedRef<HTMLDivElement>
 ) {
-	const { openInfo, important, ...muiProps } = props;
+	const {
+		openInfo,
+		important,
+		valueFormatted,
+		onClear,
+		onChange,
+		...muiProps
+	} = props;
 	const inputClasses = useInputStyles({ important });
+	const [value, setValue] = useState(muiProps.value);
+
+	const isTouchDevice = useCallback(() => {
```

Comment:
turn this into a util

Reply:
@TQjona Updated changes.. Please Review..